Modules/mathmodule.c

```diff
@@ -1724,14 +1724,19 @@ completes the proof sketch.
    Given `n` satisfying `2**62 <= n < 2**64`, return `a`
    satisfying `(a - 1)**2 < n < (a + 1)**2`. */
 
-static uint64_t
+/* Lookup table to short-circuit the first division step in _approximate_isqrt.
+   [8*round(sqrt(72+16*n)) for n in range(12)].  */
+static const uint8_t isqrt_tab[12] = {
+    64, 72, 80, 88, 88, 96, 104, 104, 112, 112, 120, 128,
+};
+
+static inline uint32_t
 _approximate_isqrt(uint64_t n)
 {
-    uint32_t u = 1U + (n >> 62);
-    u = (u << 1) + (n >> 59) / u;
-    u = (u << 3) + (n >> 53) / u;
-    u = (u << 7) + (n >> 41) / u;
-    return (u << 15) + (n >> 17) / u;
+    uint32_t u = isqrt_tab[(n >> 60) - 4];
+    u = u + (uint32_t)(n >> 50) / u;
```

Comment:
The left shift by `3` has been absorbed into the values in the lookup table, which is why we have `u` here in place of `u << 3` and `n >> 50` in place of `n >> 53`.